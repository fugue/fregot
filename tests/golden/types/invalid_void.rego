package void

test_assign_void {
  x = array.concat([1], [2], out)
  out == [1, 2]
}

test_array_void {
  all([array.concat([1], [2], out)])
}
