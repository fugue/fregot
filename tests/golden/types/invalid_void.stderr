fregot (typecheck error):
  "invalid_void.rego" (line 4, column 7):
  Void type:

    4|   x = array.concat([1], [2], out)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^

  This expression does not have a result, so it should not be assigned or used.

  "invalid_void.rego" (line 4, column 3):
  Void type:

    4|   x = array.concat([1], [2], out)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  However, it used here.

fregot (typecheck error):
  "invalid_void.rego" (line 9, column 8):
  Void type:

    9|   all([array.concat([1], [2], out)])
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^

  This expression does not have a result, so it should not be assigned or used.

  "invalid_void.rego" (line 9, column 7):
  Void type:

    9|   all([array.concat([1], [2], out)])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  However, it used here.
